### ISSUE TITLE
Cannot pass long field names in multipart/form-data

### ISSUE THREAD

**Issue by sshymko** [swoole/swoole-src#2733]
**Preconditions:**
- Swoole 4.x (any version)

**Steps to reproduce:**
1. Create a simple Swoole server that responds with parsed POST data:
    ```php
    <?php
    $server = new \Swoole\Http\Server('127.0.0.1', 8080);
    $server->set([
        'http_parse_post' => true,
    ]);
    $server->on('request', function ($request, $response) {
        $response->header('Content-Type', 'text/plain');
        $response->end(print_r($request->post, true));
    });
    $server->start();
    ```
2. Send `multipart/form-data` POST request with field name exceeding 128 characters:
    ```bash
    curl -F field1=value1 \
         -F extremely_long_field_name_exceeding_128_characters_because_some_php_frameworks_render_forms_having_unreasonably_long_field_names=value2 \
         http://127.0.0.1:8080/
    ```

**Expected result:**
- All form fields have been parsed:
    ```
    Array
    (
        [field1] => value1
        [extremely_long_field_name_exceeding_128_characters_because_some_php_frameworks_render_forms_having_unreasonably_long_field_names] => value2
    )
    ```
- Form field name length is not restricted according to [RFC 7578](https://tools.ietf.org/html/rfc7578)

**Actual result:**
- Long form field name has not been parsed and excluded from the POST data:
    ```
    Array
    (
        [field1] => value1
    )
    ```
- Error message in the standard error stream:
    ```
    [2019-08-02 12:18:00 *46081.7]	WARNING	multipart_body_on_header_value: form_name[extremely_long_field_name_exceeding_128_characters_because_some_php_frameworks_render_forms_having_unreasonably_long_field_names] is too large.
    ```
- Cookie name limit of 128 characters irrelevant to form fields is enforced
    https://github.com/swoole/swoole-src/blob/master/swoole_http_request.cc#L521

**Comment by sshymko** [comment]
RFCs do not seem to limit length of form field names. Cookie name limit is irrelevant to form fields.
There're PHP frameworks generating long form field names making them incompatible with Swoole.

**Comment by sshymko** [comment]
Until the fix is released in Swoole 4.4.4, the workaround is to parse request body `$request->rawcontent()` of `multipart/form-data` requests and append missing data to `$request->post`.
